### ISSUE TITLE
Feature Request: Emit End Column

### ISSUE THREAD

**Issue by eatnumber1** [koalaman/shellcheck#637]
Barring #636 happening, it would still be useful if ShellCheck simply emitted the end column of an issue.

One of the checks which I'm having a hard time educating users on both _how_ and _why_ it should be fixed is `SC2086`. It would go a very long way in helping if I could generate a suggested fix for it. Barring #636 happening, I can generate a suggested fix for just `SC2086` if an end column is provided.

What kinds of changes would need to happen within ShellCheck to make this happen? I'm happy to write this myself, and intend to dig into this problem over the next couple of weeks.

**Comment by brother** [comment]
Isn't the suggestions in the wiki perfectly clear in the case of SC2086? Or was that just a bad example of a error with good exemples? 
And you really want to include the wikitext into the binary? I can see why it is handy but I am not sure it's very wise in the long run.

**Comment by eatnumber1** [comment]
So there's two issues with education here:

1. Many people don't think there's an issue at all with not quoting, and telling them to quote things is stylistic at best, or worse, will actually include quotes in the value they are using.
2. Many people trip up on the fact that command expansions reset quoting context. e.x. `echo "$(echo "$foo")"`.

**Comment by brother** [comment]
Sounds more like they need education than a tool. They either do not understand or refuses to understand.

```
brother /tmp/SC2086$ touch a b c
brother /tmp/SC2086$ touch "a b c"
brother /tmp/SC2086$ ls -la
totalt 40
drwxr-xr-x  2 brother brother  4096 mar 30 16:09 .
drwxrwxrwt 41 root    root    32768 mar 30 16:09 ..
-rw-r--r--  1 brother brother     0 mar 30 16:09 a
-rw-r--r--  1 brother brother     0 mar 30 16:09 a b c
-rw-r--r--  1 brother brother     0 mar 30 16:09 b
-rw-r--r--  1 brother brother     0 mar 30 16:09 c
brother /tmp/SC2086$ for f in .; do
> echo $f
> done
.
brother /tmp/SC2086$ for f in ./; do echo $f; done
./
brother /tmp/SC2086$ for f in ./*; do echo $f; done
./a
./a b c
./b
./c
brother /tmp/SC2086$ for f in a b c; do echo $f; done
a
b
c
brother /tmp/SC2086$ for f in a b c; do echo $f; done
a
b
c
brother /tmp/SC2086$ for f in a b c; do
> if [[ -f $f ]]; then
>    echo "$f is a file"
> fi
> done
a is a file
b is a file
c is a file
brother /tmp/SC2086$ for f in "a b c"; do if [[ -f $f ]]; then    echo "$f is a file"; fi; done
a b c is a file
brother /tmp/SC2086$ ls -l a*
-rw-r--r-- 1 brother brother 0 mar 30 16:09 a
-rw-r--r-- 1 brother brother 0 mar 30 16:09 a b c
brother /tmp/SC2086$ ls -l "a "*
-rw-r--r-- 1 brother brother 0 mar 30 16:09 a b c
```

I have no suggestions than more education and more helping texts and examples. Given them a rm line that removes important stuff if not quoted properly and they might start to understand. On a related note it could be of interest to talk about the steam rm bug from some years ago - not strictly a quoting bug but very close.

**Comment by eatnumber1** [comment]
Respectfully, what does this have to do with showing the end column? I filed this feature request as it enables yet another way to educate people.

**Comment by brother** [comment]
I think you are trying to solve a problem with the incorrect tool. Not all nails need to be beaten with a hammer.

And I am not the project owner or anything really. A active user just.

**Comment by koalaman** [comment]
This is definitely a useful feature, because it would allow IDEs and such to highlight bad code rather than just point to the start of it.

There is no way currently to get the end of an AST node though, only the starting point is stored. The entire parsing code would have to be refractored to include end positions as well, and then all the checks would have to be updated to emit it. It would be quite an undertaking.

**Comment by eatnumber1** [comment]
This is implemented in #693 and #694. Currently, only SC1035 emits an end column != the start column.
